app/templates/terraform-main.tf_template

```diff
@@ -0,0 +1,142 @@
+# Configure Azure provider
+provider "azurerm" {
+  subscription_id = "${var.azure_subscription_id}"
+  client_id       = "${var.azure_client_id}"
+  client_secret   = "${var.azure_client_secret}"
+  tenant_id       = "${var.azure_tenant_id}"
+}
+
+# create a resource group if it doesn't exist
+resource "azurerm_resource_group" "rg" {
+    name = "{{host.name}}rg"
```

Comment:
We'll need some way to avoid naming conflicts between machines researchers create. To some extent the need for unique DNS name will mean the base names are unique within cloudlabs, so we might be OK just with that. Provided we don't have manually created resources in the same subscription too?